credsweeper/credentials/candidate.py

```diff
@@ -19,10 +19,12 @@ def __init__(self,
         self.ml_validation = KeyValidationOption.NOT_AVAILABLE
         self.line_data_list: List[LineData] = line_data_list if line_data_list else []
         self.patterns: List[regex.Pattern] = patterns if patterns else []
+        self.ml_probability = None
         self.rule_name: str = rule_name
         self.severity: Optional[Severity] = severity
         self.validations: List[Validation] = validations if validations else []
         self.use_ml: bool = use_ml
+        self.config = line_data_list[0].config if line_data_list else None
```

Comment:
why not to pass `config` directly to Candidate constructor?

Reply:
For some reason I thought that it would brake something in the architecture, but apparently no
Added with f6bd122